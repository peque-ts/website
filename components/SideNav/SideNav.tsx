import clsx from 'clsx';
import Link from 'next/link';
import React from 'react';

import { NavItem } from '../../lib/nav.types';

interface Props {
  items: NavItem[];
}

export const SideNav: React.VFC<Props> = ({ items }) => (
  <nav>
    <ul>
      {items.map(({ to, name, active }, index) => (
        <li key={index}>
          <Link href={to}>
            <a
              className={clsx('py-2 flex text-secondary-200 hover:text-white transition', {
                'font-semibold text-white': active,
              })}
            >
              {name}
            </a>
          </Link>
        </li>
      ))}
    </ul>
  </nav>
);
