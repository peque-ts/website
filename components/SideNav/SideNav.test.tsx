import { render, screen } from '@testing-library/react';

import { SideNav } from './SideNav';

describe('SideNav', () => {
  it('should render', () => {
    const { asFragment } = render(
      <SideNav
        items={[
          {
            active: true,
            name: 'First',
            order: 1,
            to: '/docs/1',
          },
          {
            active: false,
            name: 'Second',
            order: 2,
            to: '/docs/2',
          },
          {
            active: false,
            name: 'Third',
            order: 3,
            to: '/docs/3',
          },
        ]}
      />,
    );

    expect(screen.getAllByRole('link')).toHaveLength(3);
    expect(asFragment()).toMatchSnapshot();
  });
});
