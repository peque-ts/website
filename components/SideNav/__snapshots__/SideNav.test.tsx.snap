// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render 1`] = `
<DocumentFragment>
  <nav>
    <ul>
      <li>
        <a
          class="py-2 flex text-secondary-200 hover:text-white transition font-semibold text-white"
          href="/docs/1"
        >
          First
        </a>
      </li>
      <li>
        <a
          class="py-2 flex text-secondary-200 hover:text-white transition"
          href="/docs/2"
        >
          Second
        </a>
      </li>
      <li>
        <a
          class="py-2 flex text-secondary-200 hover:text-white transition"
          href="/docs/3"
        >
          Third
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
